packages/web3-eth-abi/src/coders/base/number.ts

```diff
@@ -43,25 +44,6 @@ function uint8ArrayToBigInt(value: Uint8Array, max: bigint): bigint {
 	return result - mask;
 }
 
-const numberLimits = new Map<string, { min: bigint; max: bigint }>();
-
-// precalculate all the limits
-for (let i = 8; i <= 256; i += 8) {
-	numberLimits.set(`uint${i}`, {
-		min: BigInt(0),
-		max: BigInt(2) ** BigInt(i) - BigInt(1),
```

Comment:
Exponentiation Operator usage issue was solved by specifying min browser versions in one of issue? 
Its supported in ECMAScript 2016 and 

Chrome 52 | Edge 14 | Firefox 52 | Safari 10.1 | Opera 39
Jul 2016 | Aug 2016 | Mar 2017 | Mar 2017 | Aug 2016

so react with older browsers is causing this problem only, and if some one specify min browsers list in react config and use on ECMAScript 2016 supporting browser there is no issue?

Reply:
Unfortunately, the issue is present with the latest as well as any older version of React . And any other framework that uses babel (the latest version of babel and any older version as well).

And this is what happens:
1. When building React app (to generate the build folder to be used for deployment), babel is used to tans-compile the code.
2. Babel will convert every `**` to `Math.pow`.
3. When browsing the page, an exception will be thrown because `Math.pow` does not work with `BigInt`.

Currently, the only way to avoid this behavior is to add configuration to `package.json` as mentioned in: https://github.com/babel/babel/issues/13109#issuecomment-826287089 and applied in https://github.com/web3/web3.js/issues/6187#issuecomment-1601223430

And this MR propose changes so the users do not need to add configurations to their `package.json`.

Reply:
I think it is better to have our own bigint pow function right now. and wait while Babel starts supporting it